src/ir/effects.h

```diff
@@ -755,6 +755,7 @@ class EffectAnalyzer {
       }
     }
     void visitArrayNew(ArrayNew* curr) {}
+    void visitArrayNewSeg(ArrayNewSeg* curr) {}
```

Comment:
Can this not trap if the segment has been dropped or something like that?

Reply:
Oh yes, good point. Also for out-of-bounds accesses.